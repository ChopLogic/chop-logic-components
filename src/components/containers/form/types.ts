import { PropsWithChildren } from 'react';
import { CommonComponentProps } from 'utils/types';

export type ChopLogicFormData = { [key: string]: unknown };

export type ChopLogicFormValidationState = [string, boolean][];

export type ChopLogicFormInput = HTMLInputElement | HTMLSelectElement;

export type ChopLogicFormInputParams = { name: string; value: unknown; valid?: boolean };

export type ChopLogicFormContextProps = {
  onChangeFormInput?: (params: ChopLogicFormInputParams) => void;
  initialValues?: ChopLogicFormData;
  resetSignal?: number;
};

export type ChopLogicFormProps = PropsWithChildren &
  CommonComponentProps & {
    columns?: number;
    initialValues?: ChopLogicFormData;
    hasReset?: boolean;
    onClickSubmit?: (data: ChopLogicFormData) => void;
    onReset?: React.FormEventHandler<HTMLFormElement>;
    onSubmit?: React.FormEventHandler<HTMLFormElement>;
  };
