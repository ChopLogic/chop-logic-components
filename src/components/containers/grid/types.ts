import { CommonComponentProps } from 'utils/types';

export type ChopLogicGridProps = CommonComponentProps & {
  columns: ChopLogicGridColumn[];
  data: ChopLogicGridItem[];
  caption?: string;
  selectable?: boolean;
  renderDataItem?: RenderDataItemCallback;
  onSelect?: (ids: string[]) => void;
};

export type ChopLogicGridColumn = {
  field: string;
  title?: string;
  component?: React.ReactElement;
  highlighted?: boolean;
  className?: string;
};

export type ChopLogicGridItem = {
  id: string;
  disabled?: boolean;
  [key: string]: unknown;
};

export type RenderDataItemCallback = (item: ChopLogicGridItem, field: string) => JSX.Element;
