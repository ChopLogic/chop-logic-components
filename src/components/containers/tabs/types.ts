import { CommonComponentProps } from 'utils/types';

export type ChopLogicTabsProps = CommonComponentProps & {
  tabs: ChopLogicTabItem[];
  defaultTabId?: string;
  mode?: ChopLogicTabsMode;
};

export type ChopLogicTabItem = {
  content: React.ReactElement;
  title: string;
  id: string;
  disabled?: boolean;
};

export type ChopLogicTabsMode = 'vertical' | 'horizontal';
