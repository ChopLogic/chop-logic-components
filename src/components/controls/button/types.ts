import { MouseEventHandler } from 'react';
import { Icon } from 'enums/icon';

export type ChopLogicButtonView = 'primary' | 'secondary' | 'danger' | 'icon';

export type ChopLogicButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  text?: string;
  onClick?: MouseEventHandler<HTMLButtonElement> | (() => void);
  view?: ChopLogicButtonView;
  disabled?: boolean;
  icon?: Icon;
  label?: string;
};
