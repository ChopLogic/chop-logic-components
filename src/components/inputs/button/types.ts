import { MouseEventHandler } from 'react';
import { CommonInputProps } from 'utils/types';

import { ChopLogicIconName } from 'components/misc/icon/Icon';

export type ChopLogicButtonView = 'primary' | 'secondary' | 'danger' | 'icon';

export type ChopLogicButtonProps = Partial<CommonInputProps> & {
  onClick?: MouseEventHandler<HTMLButtonElement> | (() => void);
  view?: ChopLogicButtonView;
  icon?: ChopLogicIconName;
  text?: string;
  extended?: boolean;
  type?: 'button' | 'reset' | 'submit';
};
