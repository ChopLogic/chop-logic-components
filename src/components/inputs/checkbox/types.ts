import { CommonInputProps } from 'utils/types';

export type ChopLogicCheckboxProps = CommonInputProps & {
  noLabel?: boolean;
  iconPosition?: 'left' | 'right';
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  defaultChecked?: boolean;
};
