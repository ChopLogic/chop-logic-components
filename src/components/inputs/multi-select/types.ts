export type MultiSelectValue = {
  id: string;
  label: string;
  selected: boolean;
};

export type ChopLogicMultiSelectProps = React.SelectHTMLAttributes<HTMLSelectElement> & {
  id: string;
  name: string;
  label: string;
  values: MultiSelectValue[];
  onSelectChange?: (values?: MultiSelectValue[]) => void;
  placeholder?: string;
};

export type MultiSelectDropdownProps = {
  values: MultiSelectValue[];
  isOpened: boolean;
  dropdownId: string;
  onClose: () => void;
  onSelect: (id: string) => void;
};
