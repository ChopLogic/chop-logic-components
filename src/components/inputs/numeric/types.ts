export type NumericValidationFunction = (input?: number) => boolean;

export type ChopLogicNumericInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  name: string;
  label: string;
  errorMessage?: string;
  validator?: NumericValidationFunction;
  onClear?: () => void;
  hasSpinButtons?: boolean;
  onSpinButtonClick?: (value?: number) => void;
};
