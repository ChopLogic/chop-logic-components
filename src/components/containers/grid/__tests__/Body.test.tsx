import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import ChopLogicGridBody from '../elements/Body';
import { ChopLogicGridColumn, ChopLogicGridItem } from '../types';

describe('ChopLogicGridBody', () => {
  const ChopLogicGridColumns: ChopLogicGridColumn[] = [
    { title: 'Col 1', field: 'field1' },
    { title: 'Col 2', field: 'field2' },
    { title: 'Col 3', field: 'field3' },
  ];

  const gridData: ChopLogicGridItem[] = [
    { id: 'row1', field1: 'Value 1', field2: 'Value 2', field3: 'Value 3', field4: 'Value 4' },
    { id: 'row2', field1: 'Value 5', field2: 'Value 6', field3: 'Value 7', field4: 'Value 8' },
    { id: 'row3', field1: 'Value 9', field2: 'Value 10', field3: 'Value 11', field4: 'Value 12' },
  ];

  const testProps = {
    data: gridData,
    columns: ChopLogicGridColumns,
    selectRowById: vi.fn(),
    deselectRowById: vi.fn(),
    selectable: true,
    selectedIds: [],
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <table>
        <ChopLogicGridBody {...testProps} />
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
