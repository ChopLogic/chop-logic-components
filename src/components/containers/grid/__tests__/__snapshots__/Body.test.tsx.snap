// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicGridBody > should match the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody
      class="sc-egkSDF frXiSn"
    >
      <tr
        class="sc-kLhKbu jokEaU"
      >
        <td
          class="sc-ivxoEo cMBTyn"
        >
          <div
            class="sc-khLCKb ciFKLI"
          >
            <input
              class="sc-dstKZu bbfSNT"
              id="select_row_row1"
              type="checkbox"
            />
            <label
              class="sc-keTIit eelhQH"
              for="select_row_row1"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 1
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 2
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 3
        </td>
      </tr>
      <tr
        class="sc-kLhKbu jokEaU"
      >
        <td
          class="sc-ivxoEo cMBTyn"
        >
          <div
            class="sc-khLCKb ciFKLI"
          >
            <input
              class="sc-dstKZu bbfSNT"
              id="select_row_row2"
              type="checkbox"
            />
            <label
              class="sc-keTIit eelhQH"
              for="select_row_row2"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 5
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 6
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 7
        </td>
      </tr>
      <tr
        class="sc-kLhKbu jokEaU"
      >
        <td
          class="sc-ivxoEo cMBTyn"
        >
          <div
            class="sc-khLCKb ciFKLI"
          >
            <input
              class="sc-dstKZu bbfSNT"
              id="select_row_row3"
              type="checkbox"
            />
            <label
              class="sc-keTIit eelhQH"
              for="select_row_row3"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 9
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 10
        </td>
        <td
          class="sc-ivxoEo cMBTyn"
        >
          Value 11
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
